docs/testing-modes.md

```diff
@@ -1,275 +1,260 @@
 ## Lincheck testing modes
 
-In this section we will take a closer look at the testing modes provided by Lincheck.
+In this section we will take a closer look at the testing modes provided by Lincheck. 
 
-TODO: why do you need stack as an example here? Why counter from the previous example is worse?
-TODO: It seems that you focus on the test structure, not on the testing modes. Let's re-think the guide structure.
-
-For an example we will test the lock-free Treiber stack algorithm. 
-In addition to the standard `push(value)` and `pop()` operations, we have also implemented a non-linearizable (thus, incorrect) `size()`, 
-which increases and decreases the corresponding `size` field following successful push and pop invocations.
+We are going to explain the structure of the test and what automatically happens under the hood for both stress and
```

Comment:
Now with the Counter example from the previous section it looks better to me. The Stack implementation was indeed just an extra distraction and was not used in any way to explain the testing modes.